cdap-notifications-api/src/main/java/co/cask/cdap/notifications/feeds/NotificationFeedManager.java

```diff
@@ -31,31 +33,32 @@
    * @throws NotificationFeedException if the feed has an incorrect structure;
    * for example, if it is missing a name, a namespace or a category
    */
-  boolean createFeed(NotificationFeed feed) throws NotificationFeedException;
+  boolean createFeed(Id.NotificationFeed feed) throws NotificationFeedException;
```

Comment:
Is it better to have the ``createFeed`` throw an AlreadyExistsException? An alternative is to use the [ConcurrentHashMap#putIfAbsent](http://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ConcurrentHashMap.html#putIfAbsent\(K,%20V\)) signature. 

A create returning false if was already present seems a little strange.